aptos-move/block-executor/src/view.rs

```diff
@@ -2299,4 +2715,295 @@ mod test {
+                ),
+                MoveTypeLayout::U64,
+            ]),
+        )]));
+        let value = Value::struct_(Struct::pack(vec![Value::struct_(Struct::pack(vec![
+            Value::u64(25),
+            Value::u64(30),
+        ]))]));
+        let state_value_4 = StateValue::new_legacy(value.simple_serialize(&layout).unwrap().into());
+        data.insert(KeyType::<u32>(4, false), state_value_4);
+        let base_view = MockStateView::new(data);
+        let versioned_map = MVHashMap::new();
+        let scheduler = Scheduler::new(10);
+        let latest_view = LatestView::<TestTransactionType, MockStateView, MockExecutable>::new(
+            &base_view,
+            ViewState::Sync(ParallelState::new(&versioned_map, &scheduler, 5, &counter)),
+            1,
+        );
+
+        assert_eq!(
+            latest_view
+                .get_resource_state_value(&KeyType::<u32>(1, false), None)
+                .unwrap(),
+            None
+        );
+        assert_eq!(
+            latest_view
+                .get_resource_state_value(&KeyType::<u32>(2, false), Some(&layout))
+                .unwrap(),
+            None
+        );
+        assert_eq!(
+            latest_view
+                .get_resource_state_value(&KeyType::<u32>(3, false), None)
+                .unwrap(),
+            Some(state_value_3.clone())
+        );
+
+        //TODO: This is printing Ok(Versioned(Err(StorageVersion), ValueType { bytes: Some(b"!0\0\0\0\0\0\0"), metadata: None }, None))
+        // Is Err(StorageVersion) expected here?
+        println!(
+            "data: {:?}",
+            versioned_map
+                .data()
+                .fetch_data(&KeyType::<u32>(3, false), 1)
+        );
+
+        let patched_value = Value::struct_(Struct::pack(vec![Value::struct_(Struct::pack(vec![
+            Value::u64(5),
+            Value::u64(30),
+        ]))]));
+        let state_value_4 =
+            StateValue::new_legacy(patched_value.simple_serialize(&layout).unwrap().into());
+        assert_eq!(
+            latest_view
+                .get_resource_state_value(&KeyType::<u32>(4, false), Some(&layout))
+                .unwrap(),
+            Some(state_value_4.clone())
+        );
+
+        let captured_reads = latest_view.take_reads();
+        assert!(captured_reads.validate_data_reads(versioned_map.data(), 1));
+        let read_set_with_delayed_fields = captured_reads.get_read_values_with_delayed_fields();
+
+        // TODO: This prints
+        // read: (KeyType(4, false), Versioned(Err(StorageVersion), Some(Struct(Runtime([Struct(Runtime([Tagged(IdentifierMapping(Aggregator), U64), U64]))])))))
+        // read: (KeyType(2, false), Versioned(Err(StorageVersion), Some(Struct(Runtime([Struct(Runtime([Tagged(IdentifierMapping(Aggregator), U64), U64]))])))))
+        for read in read_set_with_delayed_fields {
+            println!("read: {:?}", read);
+        }
+
+        // TODO: This assertion fails.
```

Comment:
Is this still failing or supposed to fail?

Reply:
I don't know, haven't touched parallel test after fixes, only sequential ones. worth cleaning them up too, and testing this then